src/histolab/tiler.py

```diff
@@ -1,21 +1,250 @@
-from abc import ABC, abstractmethod
+from abc import abstractmethod
 from typing import Tuple
 
 import numpy as np
 import sparse
+from typing_extensions import Protocol, runtime_checkable
```

Comment:
considering that 3.7 already have this 2 module in the typing std lib ... here I would use a try except statement to import them from the standard lib and if exception is raised let's import from typing_extension